src/mpyl/steps/models.py

```diff
@@ -251,7 +251,7 @@ def as_spec(self, spec_type: Type[ArtifactSpec]):
 
 
 @yaml_object(yaml)
-@dataclass()
+@dataclass(frozen=False)
```

Comment:
Tiny request: might be good to leave a comment here to explain why `frozen=False`.